example/azure-test-garden/README.md

```diff
@@ -0,0 +1,530 @@
+logs:
+
+```
+$ kubectl get seed azure -o json | jq .status
+{
+  "conditions": [
+    {
+      "lastTransitionTime": "2018-05-31T14:56:49Z",
+      "message": "all checks passed",
+      "reason": "Passed",
+      "status": "True",
+      "type": "Available"
+    }
+  ]
+}
+```
+
+# Create a Shoot cluster
+
+## Create a Project (namespace) for Shoots
+
+In this step we create a namespace in Gardener cluster to keep Shoot
+resource definitions (not the Shoot resources). A `project` in
+Gardener terminology is sipmly a namespace that holds group of Shoots,
+during this example we'll deploy a single Shoot. (Mind the extra
+labels defined in the Namespace.)
+
+```
+kubectl apply -f example/namespace-garden-dev.yaml
+```
+
+You can check the projects via `gardenctl`:
+
+```
+gardenctl target garden dev
+gardenctl ls projects
+```
+
+## Create a SecretBinding and related Secret
+
+We'll use same Azure credentials with
+`example/secret-seed-azure.yaml`, this is due to the fact that we use
+the same Azure Subscrion for the Shoot and Seed clusters. Differently
+from the Seed secret, in this one we don't need to provide
+`kubeconfig` since the Shoot cluster will be provivioned by Gardener,
+and we need to provide credentials for Route53 DNS records management.
+
+Update `example/secret-cloudprovider-azure.yaml` and place the secrets for your environment:
+* **data.subscriptionID**: you can learn this one with `az account show`
+* **data.tenantID**: from `az ad sp create-for-rbac` output as you can see above
+* **data.clientID**: from `az ad sp create-for-rbac` output as you can see above
+* **data.clientSecret**: from `az ad sp create-for-rbac` output as you can see above
+* **data.accessKeyID**: You need to add this field for Route53 records to be updated.
+* **data.secretAccessKey**: You need to add this field for Route53 records to be updated.
+
+**Note**: All of the above values must be base64 encoded. If you skip this it will hurt you later.
+
+```
+kubectl apply -f example/secret-cloudprovider-azure.yaml
+```
+
+And create the SecretBinding resource to allow Gardener use that secret.
+
+```
+kubectl apply -f example/secretbinding-cloudprovider-azure.yaml
+```
+
+Check the logs in gardener-controller-manager, there should not be any
+problems reported.
+
+## Create a CloudProfile
+
+We need to create a CloudProfile to be referred from the Shoot.
+
+```
+kubectl apply -f example/cloudprofile-azure.yaml
+```
+
+Check the logs in gardener-controller-manager, there should not be any
+problems reported.
+
+## Create the Shoot resource
+
+Update the fields in `example/shoot-azure.yaml`:
+* **spec.cloud.region**: `eastus` (this must match the seed cluster's region
+* **spec.dns.domain**: TODO: not what should i provide here and how it's relally used
+* **spec.dns.hostedZoneID**: This field not exist in the example you need
+  to add this field and place the Route53 Hosted Zone ID.
+* **spec.addons.kube-lego.email**: TODO: Not sure where is this used
```

Comment:
It's the email address used when using `kube-lego`.
See https://github.com/jetstack/kube-lego#environment-variables